### ISSUE TITLE
Clarify description of `period`

### ISSUE THREAD

**Issue by zklaus** [clix-meta/clix-meta#29]
In #20, @aulemahal noted the following about `period`:

> 1 . The allowed freq are printed in the yaml as a mapping. I.e., the output is:
```yaml
period:
  allowed:
    annual:
    seasonal:
    monthly:
  default: annual
```
> but I expected:
```yaml
period:
  allowed:
    - annual
    - seasonal
    - monthly
  default: annual
```
> Is this wanted?

This was indeed intentional. The purpose was to allow for the specification of additional information such as which months to cover.

However, I think the whole treatment of periods deserves improvement and your question shows that it is certainly less than clear at the moment.

In offline discussions, @larsbarring and I have come up with the following clarifications (@larsbarring, obviously, please correct me where I am wrong):

- Seasonal means that an index is calculated for any number of consecutive, contiguous blocks of months that cover the whole year. This includes the classical climatological seasons (["DJF", "MAM", "JJA", "SON"]), but could also be, e.g. ["NDJFMA", "MJJASO"].
- Annual means that there is one data point per year. Usually, this means an index referring to the whole year, like consecutive dry days per year, but can also refer to only a part of the year, for example, summer dry spells over "MJJAS".

In both cases it is useful to be able to specify something like `annual: MJJAS` or `seasonal: [NDJFMA, MJJASO]` and that was the thinking behind making those entries dicts instead of a simple list.

I think that probably it makes the most sense to dispense completely with the idea of "allowed" periods. After all, most of the time there either is no limitation, or the limitation stems from the natural applicability of the index, in which case we really don't need to enforce it artificially.

If you, @larsbarring and @aulemahal, agree with this, we can remove that part and focus on making the expression of the actual period clearer. What do you think?

**Comment by larsbarring** [comment]
This all goes back to the old ETCCDI specifications of the indices, where some were to be calculated per month and some per year, and yet others for both. But this has since been relaxed and/or expanded. So, yes @zklaus I agree. 

Regarding `seasonal`, I think that it is best viewed as sitting in-between `monthly`and `annual`, where the former is one value for every month of the year (no holes and no overlaps), and the former is one value per year, irrespective of whether the value represents the whole year of part of it. From our current use cases we only have examples of seasons that together covers the whole year without overlap. But one might think of overlapping seasons (e.g. some kind of moving average) or seasons that do not cover the year completely. But right now I do not see that this would change anything in practice.

**Comment by aulemahal** [comment]
I agree. In xclim we didn't really have a mechanism to "restrict" periods and I added one specifically when implementing clix-meta. But I agree that most of the time it is not useful to force such a thing, we simply let the user decide.
There is still a use for a "default" though.

**Comment by zklaus** [comment]
@larsbarring, I think we landed on `seasonal` always covering the whole year. If there is a need for overlapping seasons or seasons with gaps, the advice would be to simply calculate several annual indices independently.

@aulemahal, there is definitely a use for default! I think my suggestion would be to remove the whole `allowed` stuff (meaning both the column in the table and the dict in the yaml). Since then `default` is the only thing left, we can remove this intermediate level and just put the default period under period. This would give us something like
```yaml
period:
    monthly
```
or
```yaml
period:
    annual
```
or
```yaml
period:
    seasonal: [DJFMA, MJJ, ASON]
```
or
```yaml
period:
    annual: MAM
```
